### PR TITLE
[IMP] Base : Direct access to groups and ACLs from user

### PR DESCRIPTION
**Pull request by dbeguin**
Add 3 smart buttons with count in user form view to directly access :
- Groups
- Access Controls
- Record Rules

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
Hi @dbeguin thank to implement my feature https://github.com/odoo/odoo/pull/23452, could you add me in author ?
Regards

**Comment by dbeguin** [comment]
Hi @fmdl,
Thank you for your nice work ! Yes, I will add you as author. I'm just waiting for the feature to be tested and validated after the little modifications that have been made.

Kind regards

File: odoo/addons/base/models/res_users.py [hunk 1]
```diff
@@ -676,6 +680,55 @@ def _has_group(self, group_ext_id):
     # for a few places explicitly clearing the has_group cache
     has_group.clear_cache = _has_group.clear_cache
 
+    def _get_access_count(self):
+        for user in self:
+            groups = user.groups_id
+            user.access_count = len(groups.mapped('model_access'))
+            user.rules_count = len(groups.mapped('rule_groups'))
+            user.groups_count = len(user.mapped('groups_id'))
```
Comment (fmdl) [inline odoo/addons/base/models/res_users.py]: can be replace by `user.groups_count = len(groups)`

**Comment by fmdl** [comment]
@dbeguin maybe add in the name of view : `Acces right (edit for all users)`, like that the user understand that he edits for all user and not for the user in the previous view.

**Comment by fmdl** [comment]
any news ?

**Comment by dbeguin** [comment]
Hi @fmdl, 
Discussions are still ongoing on that topic. It takes so long because of the v12 freeze. But as it's now behind us, we will take it back asap. Don't worry, we don't forget you ;).

File: odoo/addons/base/models/res_users.py [hunk 2]
```diff
@@ -697,6 +701,57 @@ def _has_group(self, group_ext_id):
     # for a few places explicitly clearing the has_group cache
     has_group.clear_cache = _has_group.clear_cache
 
+    def _get_access_count(self):
```
Comment (tde-banana-odoo) [inline odoo/addons/base/models/res_users.py]: Guidelines: compute method should be prefixed by _compute. You should also add the triggers for the computation.

File: odoo/addons/base/models/res_users.py [hunk 3]
```diff
@@ -251,6 +251,10 @@ def _get_company(self):
     name = fields.Char(related='partner_id.name', inherited=True, readonly=False)
     email = fields.Char(related='partner_id.email', inherited=True, readonly=False)
 
+    access_count = fields.Integer(compute='_get_access_count')
```
Comment (tde-banana-odoo) [inline odoo/addons/base/models/res_users.py]: Please always give a string to the fields.

File: odoo/addons/base/models/res_users.py [hunk 4]
```diff
@@ -697,6 +701,57 @@ def _has_group(self, group_ext_id):
...
+    def _get_access_count(self):
+        for user in self:
+            groups = user.groups_id
+            user.access_count = len(groups.mapped('model_access'))
+            user.rules_count = len(groups.mapped('rule_groups'))
+            user.groups_count = len(groups)
+
+    def action_show_groups(self):
+        self.ensure_one()
+        self.user_has_groups('base.group_erp_manager')
```
Comment (tde-banana-odoo) [inline odoo/addons/base/models/res_users.py]: user_has_groups returns a boolean. This line has therefore no effect. I suppose you want to raise or return something ? Is it necessary as it should be covered by access rights ?

File: odoo/addons/base/models/res_users.py [hunk 5]
```diff
@@ -697,6 +701,57 @@ def _has_group(self, group_ext_id):
...
+            'res_model': 'res.groups',
+            'type': 'ir.actions.act_window',
+            'context': {'create': False, 'delete': False},
+            'domain': [('id','in', self.groups_id.ids)],
+            'target': 'current',
+        }
+
+    def action_show_access(self):
+        self.ensure_one()
+        self.user_has_groups('base.group_erp_manager')
```
Comment (tde-banana-odoo) [inline odoo/addons/base/models/res_users.py]: Same as above

File: odoo/addons/base/models/res_users.py [hunk 6]
```diff
@@ -697,6 +701,57 @@ def _has_group(self, group_ext_id):
...
+            'res_model': 'ir.model.access',
+            'type': 'ir.actions.act_window',
+            'context': {'create': False, 'delete': False},
+            'domain': [('id', 'in', self.mapped('groups_id.model_access').ids)],
+            'target': 'current',
+        }
+
+    def action_show_rules(self):
+        self.ensure_one()
+        self.user_has_groups('base.group_erp_manager')
```
Comment (tde-banana-odoo) [inline odoo/addons/base/models/res_users.py]: Same as above

File: odoo/addons/base/models/res_users.py (hunk 3 above)
Comment (tde-banana-odoo) [reply odoo/addons/base/models/res_users.py]: Maybe rename to accesses_count  to have all fields to plural mode ?

File: odoo/addons/base/models/res_users.py [hunk 7]
```diff
@@ -697,6 +701,57 @@ def _has_group(self, group_ext_id):
...
+            'view_type': 'form',
+            'view_mode': 'tree,form',
+            'res_model': 'res.groups',
+            'type': 'ir.actions.act_window',
+            'context': {'create': False, 'delete': False},
+            'domain': [('id','in', self.groups_id.ids)],
+            'target': 'current',
+        }
+
+    def action_show_access(self):
```
Comment (tde-banana-odoo) [inline odoo/addons/base/models/res_users.py]: Maybe rename to action_show_accesses to have everything using the plural form ?

File: odoo/addons/base/views/ir_model_views.xml [hunk 8]
```diff
@@ -530,6 +530,23 @@
 
         <!-- ir.model.access -->
         <record id="ir_access_view_tree" model="ir.ui.view">
+            <field name="model">ir.model.access</field>
```
Comment (tde-banana-odoo) [inline odoo/addons/base/views/ir_model_views.xml]: Field name is missing.

**Comment by tde-banana-odoo** [comment]
COuld you r+ @robodoo  ?

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**